src/values/AbstractValue.js

```diff
@@ -664,9 +667,21 @@ export default class AbstractValue extends Value {
     let values = ValuesDomain.joinValues(realm, left, right);
     let [hash, args] = hashTernary(condition, left || realm.intrinsics.undefined, right || realm.intrinsics.undefined);
     let Constructor = Value.isTypeCompatibleWith(types.getType(), ObjectValue) ? AbstractObjectValue : AbstractValue;
-    let result = new Constructor(realm, types, values, hash, args, ([c, x, y]) => t.conditionalExpression(c, x, y), {
-      kind: "conditional",
-    });
+    let result = new Constructor(
+      realm,
+      types,
+      values,
+      hash,
+      args,
+      ([c, x, y]) => {
```

Comment:
This would still serialize [c, x, y], possibly leaving around various artifacts, even if c, x or y are not used in the end.

Reply:
Yes, I have noted this in the initial comment.